library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c

```diff
@@ -378,7 +379,13 @@ static libspdm_return_t libspdm_try_negotiate_algorithms(libspdm_context_t *spdm
                      sizeof(spdm_algorithms_response_t) +
                      sizeof(uint32_t) * spdm_response->ext_asym_sel_count +
                      sizeof(uint32_t) * spdm_response->ext_hash_sel_count);
+        alg_type_pre = struct_table->alg_type;
         for (index = 0; index < spdm_response->header.param1; index++) {
+            /* AlgType shall monotonically increase for subsequent entries. */
```

Comment:
1. `alg_type_pre` needs to be updated inside the for-loop.
2. This code should be moved to the `-=[Validate Response Phase]=-` section at line `249`.

Reply:
1. I made a mistake of omitting alg_type_pre update at line 389. To prevent this mistake, I will add  not only redundant case but also decreasing case.
2. I'll move this checker to line 249.